### PR TITLE
CA-362358: Filter out new files when refreshing directory of certific…

### PR DESCRIPTION
**Pull request by snwoods**
…ates

Signed-off-by: Steven Woods <steven.woods@citrix.com>

### CODE REVIEW

**Comment by psafont** [comment]
There's at least one other place that needs to do a better job of filtering this directory:

https://github.com/xapi-project/xen-api/blob/master/ocaml/xapi/cert_distrib.ml#L460

**Comment by lindig** [comment]
I think this looks good. I speculate the problem of suffix clashes could have also been resolved by creating new certificates under `*.pem.new` rather than `*.new.pem`.

**Review by lindig — APPROVED** [review]

File: ocaml/xapi/cert_distrib.ml [hunk 1]
```diff
@@ -457,7 +457,10 @@ let ( (get_local_ca_certs : unit -> WireProtocol.certificate_file list)
      * executed on [path] *)
     Sys.readdir path
     |> Array.to_list
-    |> List.filter (Fun.flip Filename.check_suffix "pem")
+    |> List.filter (fun x ->
```
Comment (lindig) [inline ocaml/xapi/cert_distrib.ml:460]: Minor detail: a less generic name than `x` would be better - something like `file`, `fname`, `name`, or similar. Below `filename` is used.

**Review by psafont — APPROVED** [review]
Once we see there are no more bugs blocking distribution of builds we can merge this

**Pull request merged**